bytearray_operations.go

```diff
@@ -44,6 +44,83 @@ func LeftShift(data []byte, shift uint64) []byte {
 	return result
 }
 
+// Mask apply AND mask to byte array
+func Mask(data, mask []byte) []byte {
+	var dataLength = len(data)
+	result := make([]byte, dataLength)
+	if dataLength < 1 {
+		return result
+	}
+	copy(result, data)
+
+	var maskLength = len(mask)
+	var operationLength = maskLength
+	// If mask is longer than data, keep operation to data length
+	if maskLength > dataLength {
+		operationLength = dataLength
+	}
+	for i := 0; i < operationLength; i++ {
+		result[i] = data[i] & mask[i]
+	}
+
+	return result
+}
+
+// InclusiveMerge apply OR mask to byte array
+func InclusiveMerge(data, mask []byte) []byte {
```

Comment:
It's not data and mask, but 2 datas that will be merged with OR operations.
If 2 datas have not the same length, the biggest length win and the missing byte of the shortest array will be equals to `0x00`